src/IR/LowerAst.hs

```diff
@@ -102,7 +108,8 @@ propogate it elsewhere. This is possible because the IR's type annotations
 the join operation.
 -}
 lowerExpr :: A.Expr -> AnnotationK -> I.Expr I.Type
-lowerExpr (  A.Id  v    ) k = I.Var (fromString v) (k I.untyped)
+lowerExpr (A.Id v) k | isCons v  = I.Data (fromId v) (k I.untyped)
```

Comment:
Here we differentiate between data constructors and variables, and lower to different IR nodes.